packages/common-types/src/base/Point.ts

```diff
@@ -32,8 +32,7 @@ class Point implements IPoint {
         return Buffer.concat([Buffer.from("0x04", "hex"), Buffer.from(this.x.toString("hex"), "hex"), Buffer.from(this.y.toString("hex"), "hex")]);
       case "elliptic-compressed": {
         // TODO: WHAT IS THIS.?
-        let ec = params;
-        ec = ecCurve;
+        const ec = params?.ec || getEllipticCurve("secp256k1");
```

Comment:
ec or keytype should be part of the Point's member

Reply:
done

Reply:
add the curve as well, *initialize the curve on constructor using the keyType
https://github.com/torusresearch/torus.js/blob/f6f9afc3a26cec4e26b059de342d887a10f1e220/src/Point.ts#L11